### PR TITLE
Added the requirements analysis for the changes needed to SODA Dashboard for Isabela release

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind design

**What this PR does / why we need it**:
This PR contains the requirements and analysis for the changes and features needed in the SODA dashboard for the I and subsequent releases.
The requirements cover the changes needed in :
- Multicloud UI
- Delfin UI
- Multicloud HA (depends on changes needed )
- Overall bug fixes and improvements.

**Which issue(s) this PR fixes**:
Fixes #

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

File: arch-design/dashboard/requirements_analysis_for_i_release.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+The handling of buckets on cloud storage using SODA Multicloud APIs is undergoing a design change as per this pr [architecture-analysis/99](https://github.com/sodafoundation/architecture-analysis/pull/99). This will change the fundamental design of how SODA Multicloud deals with registration of backends and buckets. This requires some modifications in API parameters and some changes to how the Dashboard lists and operates on backends, buckets and objects within those buckets. SODA Delfin is adding new drivers support on a regular basis and this requires the Dashboard to be updated to allow the user to register storages for these new drivers.Additionally any bug fixes and improvements will be covered as part of this release
+
+
+### Requirement Analysis
+
+
+#### Multicloud - Bucket Management
+---
+* Create bucket on cloud backend (AWS, Azure, GCP and all supported vendors) based on custom tier policy
```
Comment (kumarashit) [inline arch-design/dashboard/requirements_analysis_for_i_release.md:45]: These are mainly the functional requirements. Requirement should be high level

File: arch-design/dashboard/requirements_analysis_for_i_release.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+##### Non Functional Requirements
+
+  
+## Architecture Analysis
+
+### Module Architecture
+
+
+The current SODA Multi-cloud architecture can be seen in Fig-1 below.
```
Comment (kumarashit) [inline arch-design/dashboard/requirements_analysis_for_i_release.md:112]: 1) This doesn't seem to be architecture but mainly flow
2) What is the difference between soda-bucket and cloud bucket?
3) So the diff between two is just the step 1), correct?
4) Does step 2 follows step 1?

**Review by kumarashit — CHANGES_REQUESTED** [review]

File: arch-design/dashboard/requirements_analysis_for_i_release.md (hunk 2 above)
Comment (anvithks) [reply arch-design/dashboard/requirements_analysis_for_i_release.md:112]: > 1. This doesn't seem to be architecture but mainly flow

The main architecture has not changed for the Dashboard. You are right. This is the flow that is depicted here. I can link to the architecture diagram [here](https://github.com/sodafoundation/architecture-analysis/blob/d1dfdc6663534c4eab3eba3f235f8b09c4362c4a/arch-design/dashboard/resources/angular-app-architecture.png)
> 2. What is the difference between soda-bucket and cloud bucket?

In Fig -1 cloud-bucket and soda-bucket makes sense. The existing SODA multilcoud architecture requires the user to create bucket on the cloud using the cloud console and provide the bucket name as an input during the Backend registration process in SODA. The soda-bucket refers to all the buckets created within SODA on the registered backends. These are in fact folders that are created within the cloud-buckets.
After the changes that have been proposed are implemented the step to create a cloud bucket can be performed through SODA multicloud itself. Here there would be no need for the user to create a bucket and provide the name as part of backend registration. During Iter2_I the backend registration was not changed and that is why this has remained here. 
Will be fixed.

> 3. So the diff between two is just the step 1), correct?

Correct. In the new changes there will be no difference in cloud-bucket and soda-bucket. All buckets that user creates from SODA will be actual buckets and not folders before.

> 4. Does step 2 follows step 1?

This is slightly confusing. These numbers do not signify order (although numbers usually do.)
I will fix these and update the diagrams. 

@PravinRanjan10 we should look at the diagrams for the multicloud sequence diagrams again and make the changes discussed above.

File: arch-design/dashboard/requirements_analysis_for_i_release.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+SODA Delfin is adding new drivers support on a regular basis and this requires the Dashboard to be updated to allow the user to register storages for these new drivers.Additionally any bug fixes and improvements will be covered as part of this release
+
+
+### Requirement Analysis
+
+
+#### Multicloud - Bucket Management
+---
+- **Register Backend**  
+    The current requirement of creating a bucket on the cloud backend and providing that bucket name for backend registration in SODA Multicloud will be removed for AWS, Azure and GCP backends. User will provide all the other parameters as before for backend registration.
```
Comment (kumarashit) [inline arch-design/dashboard/requirements_analysis_for_i_release.md:49]: only for AWS, Azure, GCP?

Comment (anvithks) [reply arch-design/dashboard/requirements_analysis_for_i_release.md:49]: Updated with details of supported clouds.

File: arch-design/dashboard/requirements_analysis_for_i_release.md (hunk 1 above)
Comment (anvithks) [reply arch-design/dashboard/requirements_analysis_for_i_release.md]: I think the high level requirements are listed in the Goals section. This is the Analysis section so there are some more details.

File: arch-design/dashboard/requirements_analysis_for_i_release.md (hunk 2 above)
Comment (anvithks) [reply arch-design/dashboard/requirements_analysis_for_i_release.md:199]: @kumarashit This has been updated. Please check the updated diagrams. These have been added to the document as well.

Current .
multi_cloud_architecture_current

Updated.
multi_cloud_architecture_updated

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM
(some parts may need refinement, which could be taken for the next release...like grafana part etc)

**Pull request merged**